pyptex/__init__.py

```diff
@@ -229,298 +241,305 @@ def genname(self,pattern="fig{gencount}.eps"):
           that `a.pyptex` is correctly compiled into `a.pdf` and that a stale cache is
           never used.
-        * `pyp.deps` is a dictionary of dependencies and timestamps
-        * `pyp.lc` counts lines while parsing
-        * `pyp.argv` stores the ``command-line arguments'' for template generation
-        * `pyp.exitcode` is the exit code of the `pyp.latexcommand`
-        * `pyp.gencount` is the counter for generated files (see `pyp.gen()`)
-        * `pyp.fragments` is the list of Python fragments extracted from a.tex
+        * `pyp.deps` is a dictionary of dependencies and timestamps.
+        * `pyp.lc` counts lines while parsing.
+        * `pyp.argv` stores the ``command-line arguments'' for template generation.
+        * `pyp.exitcode` is the exit code of the `pyp.latexcommand`.
+        * `pyp.gencount` is the counter for generated files (see `pyp.gen()`).
+        * `pyp.fragments` is the list of Python fragments extracted from a.tex.
         * `pyp.outputs` is the matching outputs.
-        * `pyp.compiled` is the string that is written to `a.pyptex`
+        * `pyp.compiled` is the string that is written to `a.pyptex`.
         """
-        print(texfilename+": pyptex compilation begins")
-        self.__globals__ = {"__builtins__":__builtins__,"pyp":weakref.proxy(self)}
+        print(f'{texfilename}: pyptex compilation begins')
+        self.__globals__ = {'__builtins__': __builtins__, 'pyp': weakref.proxy(self)}
         self.filename = os.path.splitext(texfilename)[0]
         self.texfilename = texfilename
-        self.cachefilename = self.filename+'.pickle'
-        self.bibfilename = self.filename+'.bib'
-        self.pyptexfilename = self.filename+'.pyptex'
-        self.auxfilename = self.filename+'.aux'
-        self.latex = "pdflatex --file-line-error --synctex=1"
+        self.cachefilename = f'{self.filename}.pickle'
+        self.bibfilename = f'{self.filename}.bib'
+        self.pyptexfilename = f'{self.filename}.pyptex'
+        self.auxfilename = f'{self.filename}.aux'
+        self.latex = 'pdflatex --file-line-error --synctex=1'
         self.latexcommand = latexcommand
         self.disable_cache = False
         self.deps = {}
         self.lc = 0
-        self.argv = argv
+        self.argv = [] if argv is None else []
         self.__sympy_plot__ = None
         self.exitcode = 0
         self.generateddir()
         self.dep(__file__)
         self.compile()
-        print(texfilename+": pyptex compilation ends")
-
+        print(f'{texfilename}: pyptex compilation ends')
 
-    def run(self,S,k):
-        """An internal function for executing Python code"""
-        print("Executing Python code:\n"+S)
-        S = "\n"*k+S
-        glob = self.__globals__
+    def run(self, S, k):
+        """An internal function for executing Python code."""
+        print(f'Executing Python code:\n{S}')
+        S = '\n'*k + S
+        glob_ = self.__globals__
```

Comment:
Why the underscore? What is this fixing?

Reply:
The `glob` name conflicts with the `glob` module that was imported. Shadowing the name could lead to bugs if anything is ever added after that where you could think you're using the `glob` module when you're actually using `self.__globals__`. Shadowing like that also causes PyFlakes warning [F811](http://flake8.pycqa.org/en/2.5.5/warnings.html).